gitpod-io/gitpod
# Proposal: feat(gitpod-cli): Enable completion command and unhide it

## Description

Glad I stumbled upon the codebase, I was literally about to write the completion scripts for use with my fish shell. It's kind of funny how this wasn't mentioned anywhere but it was already there for `bash` 😬 `gp` CLI got lots of subcommands and options, having tab-completion is super helpful, always missed it.

## Related Issue(s)

Fixes #

## How to test

1. Open a workspace in preview env
2. Type `gp` and hit `Tab` key.
- To check for other shells, just switch to them by `fish` or `zsh`
- And redo `2.`

## Release Notes

```release-note
Tab-completions of `gp` CLI is now auto installed for bash, fish and zsh
```

## Documentation
- [ ] Have to document as well


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`
